### PR TITLE
[ML] Ensure unique IDs between inference models and trained model deployments

### PR DESCRIPTION
**Pull request by maxhniebergall**
The inference service uses trained model deployments to access in-cluster models. The inference service ID for these models is the inference service `modelId`, which is mapped to `deploymentId` in the trained models API. To avoid bugs due to conflicting IDs, this change restricts the `PutInferenceModelAction` and the `StartTrainedModelDeploymentAction` to unique IDs.

I also refactored the ml-inference-service-tests integration tests to use a shared base class which contains methods used by multiple subclasses.  

closes https://github.com/elastic/ml-team/issues/1094

### CODE REVIEW

File: x-pack/plugin/inference/build.gradle [hunk 1]
```diff
@@ -20,7 +20,8 @@ base {
 
 dependencies {
   implementation project(path: ':libs:elasticsearch-logging')
-  compileOnly project(":server")
+    implementation project(path: ':x-pack:plugin:core')
```
Comment (mark-vieira) [inline x-pack/plugin/inference/build.gradle:23]: nit: can we fix the indenting here?

**Review by mark-vieira — APPROVED** [review]

File: x-pack/plugin/ml/qa/ml-inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ModelIdUniquenessIT.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    public void testPutTrainedModelFailsWhenInferenceModelWithIdAlreadyExists() throws Exception {
+
+        String modelId = "duplicate_model_id";
+        putPyTorchModelTrainedModels(modelId);
+        putPyTorchModelDefinitionTrainedModels(modelId);
+        putPyTorchModelVocabularyTrainedModels(List.of("these", "are", "my", "words"), modelId);
+        startDeploymentTrainedModels(modelId);
+
+        var e = expectThrows(ResponseException.class, () -> putInferenceServiceModel(modelId, TaskType.SPARSE_EMBEDDING));
```
Comment (davidkyle) [inline x-pack/plugin/ml/qa/ml-inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ModelIdUniquenessIT.java:72]: This looks like the same test as `testPutInferenceModelFailsWhenTrainedModelWithIdAlreadyExists`, shouldn't this one PUT the inference service model first?

File: x-pack/plugin/ml/qa/ml-inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ModelIdUniquenessIT.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.hamcrest.Matchers;
+import org.junit.ClassRule;
+
+import java.util.List;
+
+public class ModelIdUniquenessIT extends InferenceBaseRestTest {
+
+    @ClassRule
```
[the remark this answers is not in the document]
Comment (maxhniebergall) [reply x-pack/plugin/ml/qa/ml-inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ModelIdUniquenessIT.java]: Thanks!

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**